lib/excon/connection.rb

```diff
@@ -32,6 +32,16 @@ def proxy=(new_proxy)
       @data[:proxy] = new_proxy
     end
 
+    def log
+      if @data[:instrumentor] && @data[:instrumentor].respond_to?(:logger)
+        @data[:instrumentor].logger
+      end
+    end
+    def log=(logger)
```

Comment:
Maybe this one should be `logger=`? What do you think?

Reply:
I didn't want to confuse people if they wanted to use some other logging mechanism.

Reply:
Yeah, I guess both `log` and `log=` seemed weird to me if they call logger under the covers. ie Are you getting back a thing to push logs to, or the logs themselves? Does that make sense?

Reply:
Ok, I'll update this.

Reply:
Thanks!

Reply:
Ok, updated the method names, and PR description to go with it. Also, I noticed a bug in the way query params were handled, so I updated that as well.